format animal ids for custom tasks

antsgar opened the pull request:
**Description**

Tested #3562 and #3565 together, this change in the format between what's saved by the form and what's sent to the API is required for custom task creation to work with related animals.

Jira link:

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

kathyavini reviewed (commented):
I did exactly line for line the same thing 😂 

But then I realized you can also call in the main body of the saga like this:

```js
    data = getCompleteCustomTaskTypeBody(data, task_translation_key);

    if (data.animalIds) {
      data = formatAnimalTask(data, endpoint);
    }
   ```
   
and not touch this produce function, which I think reads a little cleaner. But I'm not sure if I'm using @SayakaOno's functions right when I do that, so I'll definitely defer to her 😁 
 
 I think the code that runs is identical!

SayakaOno commented:
`formatAnimalTask` removes `managementPlans`, which are allowed for custom tasks. Unfortunately, this means `formatAnimalTask` can't be used there.

But the ID formatting will happen twice for movement tasks, so should this function be combined with `getPostTaskBody`?
```
export const formatAnimalTask = (data, endpoint) => {
  const { animalIds, managementPlans, ...formattedData } = getPostTaskBody(data, endpoint);

  // Remove managementPlans from the data and format animalIds for the request body
  return { ...formattedData, ...formatAnimalIdsForReqBody(animalIds) };
};`
```
What's left is not to add `managementPlans` for animal tasks.

antsgar commented:
@kathyavini @SayakaOno I think I prefer to keep the main body of the saga as decluttered as possible (although all of this sage is pretty cluttered at the moment 😅), but there was definitely redundancy between `formatAnimalTask` and the changes here. I removed `formatAnimalTask` so that these changes would be responsible for handling formatting the animal IDs for all tasks with them (including Movement) and just added the removal of management plans to the movement task formatting function instead. Let me know if that works, or if I'm missing something!
